### PR TITLE
Integrate Gravatar SDK

### LINKED ISSUE #22543: [Gravatar] Gravatar SDK integration umbrella task
Let's list and track work items related with Gravatar integration here.

Replace avatar fetching calls with SDK calls in following places, make sure each flow is tested:

- [x] CommentContentTableViewCell
- [x] MyProfileHeaderView
- [x] NoteBlockCommentTableViewCell
- [x] EpilogueUserInfoCell
- [x] AuthorFilterButton
- [x] AuthorFilterViewController
- [x] ListTableViewCell
- [x] GravatarEmailTableViewCell
- [x] LoginLinkRequestViewController
- [x] UIBarButtonItem+MeBarButton

---

Investigate following methods in the existing UIImageView+Gravatar extension, make sure we don't break the functionality for them after integrating the SDK

- [x] overrideGravatarImageCache
- [x] listenForGravatarChanges

---

- [x] Profile fetching: We won't replace this, the code is unreachable.

Task list for profile uploading:

- [x] Avatar uploading

---

Task list for rename types in Gravatar SDK when `Gravatar` is used unnecessarily:
- [ ] `ImageCache` and `ImageCaching`

### PR DESCRIPTION
**Pull request by pinarol**
Fixes https://github.com/wordpress-mobile/WordPress-iOS/issues/22543

WordPressUI PR: https://github.com/wordpress-mobile/WordPressUI-iOS/pull/156

- Adds a new dependency: [Gravatar](https://github.com/Automattic/Gravatar-SDK-iOS)
- Replaces `WordPressUI.Gravatar` usage with `Gravatar.AvatarURL`
- Starts using `Gravatar.AvatarService.upload(...)` to upload an avatar
- WordpressUICacheAdapter now also conforms to `Gravatar.ImageCaching`.
- Adds `extension UIImageView` to add new alternatives of what’s deprecated in WordPressUI. (All related with downloading an Avatar.)
- Updates `GravatarServiceTests`

No new functionality is added. This is just to replace code & refactor. 

------

I'll add some screenshots with avatars to make `dangermattic` happy.

------

To test:

- [x] [Test cases for downloading avatar via email](https://github.com/wordpress-mobile/WordPress-iOS/issues/22543#issuecomment-1953982734)
- [x] [Test cases for downloading avatar via URL](https://github.com/wordpress-mobile/WordPress-iOS/issues/22543#issuecomment-1948014118)
- [x] [Test cases for avatar upload](https://github.com/wordpress-mobile/WordPress-iOS/issues/22543#issuecomment-1956744840)

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [x] WordPress.com sites and self-hosted Jetpack sites.
- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [x] High contrast.
- [x] VoiceOver.
- [x] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] iPhone and iPad. 
- [x] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

File: Podfile [hunk 1]
```diff
@@ -41,13 +41,18 @@ def aztec
...
+  # pod 'WordPressUI', '~> 1.15'
   # pod 'WordPressUI', git: 'https://github.com/wordpress-mobile/WordPressUI-iOS', tag: ''
   # pod 'WordPressUI', git: 'https://github.com/wordpress-mobile/WordPressUI-iOS', branch: ''
-  # pod 'WordPressUI', git: 'https://github.com/wordpress-mobile/WordPressUI-iOS', commit: ''
+  pod 'WordPressUI', git: 'https://github.com/wordpress-mobile/WordPressUI-iOS', commit: '7f7606eefceaaab9982cf3417093a45d78bf284d'
   # pod 'WordPressUI', path: '../WordPressUI-iOS'
 end
 
+def gravatar
+  #pod 'Gravatar', path: '../Gravatar-SDK-iOS'
```
Comment (dangermattic) [inline Podfile:52]: <table data-meta="generated_by_pr-check">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/wordpress-mobile/WordPress-iOS/blob/000e6da08db07b70f06233492283d1642cafd071/Podfile#L52"></span>Layout/LeadingCommentSpace: Missing space after <code>#</code>.</td>
    </tr>
  </tbody>
</table>

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by pinarol** [comment]
cc: @alpavanoglu @kean Could you take a look as well? Not sure who else to tag so please feel free to tag others who might be interested. We don't expect you to run the tests(although you are highly welcome if you want to). Just let us know if you have any concerns & comments. Thank you!

**Comment by pinarol** [comment]
Also, here's the alternative change set that imports Gravatar from WordPressUI:

https://github.com/wordpress-mobile/WordPress-iOS/compare/task/gravatar-integration?expand=1
https://github.com/wordpress-mobile/WordPressUI-iOS/compare/task/gravatar-integration?expand=1

The reason why we didn't pursue that is because WordPressUI has deployment target set to iOS 11 and this is too low for our SDK. Gravatar SDK’s is iOS 15. For us, it’s better not to be bound by WPUI’s target version as we go down the road. Because we’ll add more features in the SDK and we would like to make the most of modern Apple APIs. So it would be easier for us to move fwd with this PR. 

But let us know if you think this one is better.

**Review by etoledom — APPROVED** [review]
All looks good to me!

Smoke-tested using the test cases from https://github.com/wordpress-mobile/WordPress-iOS/issues/22543

✅

**Comment by pinarol** [comment]
> But let us know if you think this one is better.

We talked in Slack. We decided to move fwd with this solution.

**Pull request merged**